common/common.go

```diff
@@ -83,14 +83,47 @@ func LoadBuildConfigFromProvenance(provenance *slsa.Provenance) (*BuildConfig, e
 		return nil, fmt.Errorf("the provenance must have exactly one Subject, got %d", len(provenance.Subject))
 	}
 
-	proParams := provenance.Predicate.Invocation.Parameters
+	expectedBinarySha256Hash := provenance.Subject[0].Digest["sha256"]
+	if len(provenance.Subject) != 1 {
+		return nil, fmt.Errorf("the provenance's subject digest must specify a sha256 hash, got %d", expectedBinarySha256Hash)
+	}
+
+	if len(provenance.Predicate.Materials) != 2 {
+		return nil, fmt.Errorf("the provenance must have exactly two Materials, got %d", len(provenance.Predicate.Materials))
+	}
+
+	builderImage := provenance.Predicate.Materials[0].URI
```

Comment:
I was hoping that validating at this level of detail would not be needed :) 
I mean if the code does not panic when accessing `provenance.Predicate.Materials[0].URI` if `provenance.Predicate.Materials[0]` is null, then we can just use `provenance.Predicate.Materials[0].URI` without having to validate it. But it also depends on our definition of a valid provenance. 

Let's refine this in a separate PR.

Reply:
Fair :). I'm not sure how go works exactly, does it panic if the material is not defined? My understanding was it just defaults to an empty string :)

Reply:
Looked it up, seems like it indeed would just panic at this point. :) Yh we don't need that check then.

Reply:
Yeah, I was also not sure how exactly go works in these situations. But if it panics, then we'd need a check to avoid the panic. No?